src/pykeen/pipeline/api.py

```diff
@@ -1052,6 +1052,8 @@ def pipeline(  # noqa: C901
             " (Bordes et al., 2013).",
         )
         additional_filter_triples.append(validation.mapped_triples)
+    elif evaluator_instance.filtered and isinstance(stopper, EarlyStopper):
```

Comment:
It would be good to add a warning when the users has set `filter_validation_when_testing=False`

Reply:
this code shouldn't even be touched unless `evaluation.filtered` is happening, I will wrap all of it in an if statement then come back to this.

Reply:
While not a very common practice for KG link prediction (afaik), you sometimes see the final best configuration trained again on `train+validation`, before evaluating on the test set. In this case, we would not have a set of validation triples.

So maybe an information would be a more appropriate logging level?

Reply:
Check the new code, I added a logging message but it was before your suggestion

Reply:
@mali-git I added this check to require filter_validation_when_testing=True in f021a29e. I don't think it needs a logging message - it can work the same as for HPO